src/olympia/users/models.py

```diff
@@ -515,17 +516,30 @@ def ban_and_disable_related_content_bulk(cls, users, move_files=False):
             'picture_type', 'banned', 'biography', 'display_name', 'homepage',
             'location', 'deleted', 'auth_id', 'username'))
 
-    def delete(self, hard=False, addon_msg=''):
-        if hard:
-            # this is done in _anonymize() for soft deletes
-            self.picture_type = None
-            self.delete_picture()
-            super(UserProfile, self).delete()
-        else:
-            self._delete_related_content(addon_msg=addon_msg)
-            log.info(f'User ({self}: <{self.email}>) is being anonymized.')
-            self._anonymize()
-            self.save()
+    def _prepare_delete_email(self):
+        email_to = [self.email]
+        site_url = settings.EXTERNAL_SITE_URL
+        template = loader.get_template('users/emails/user_deleted.ltxt')
+        email_msg = template.render(context={
+            'site_url': site_url,
+            'name': self.name})
+        return {
+            'subject': f'Your account on {site_url} has been deleted',
+            'message': email_msg,
+            'recipient_list': email_to,
+        }
+
+    def delete(self, addon_msg='', send_delete_email=True):
```

Comment:
No more `hard` param?

Reply:
We didn't use it anywhere so I took the opportunity to remove a code path.  (we could still do `UserProfile.objects.get(...).delete()` if we particularly wanted to hard delete in a test)